front/lib/auth.ts

```diff
@@ -102,7 +102,20 @@ export class Authenticator {
     this._key = key;
   }
 
-  static aclsFromGroupIds(groupIds: string[]): ACLType[] {
+  /**
+   * Converts an array of group sIDs (string identifiers) into ResourcePermission objects.
+   * This utility method creates standard read/write permissions for each group.
+   *
+   * The resulting permissions enforce a conjunction (AND) between groups. A user must belong to
+   * ALL groups to access the resource, as each group is placed in a separate ResourcePermission
+   * entry.
+   *
+   * @param groupIds - Array of group string identifiers
+   * @returns Array of ResourcePermission objects, one per group, requiring membership in all groups
+   */
+  static resourcePermissionsFromGroupIds(
```

Comment:
nit: `createResourcePermissionsFromEachGroupIds(..)` and move it elswhere (it's confusing, I had to look carefuly to understand it's completely unrelated to the group of the Authenticator).